download/tests/download-curl-resume.rs

```diff
@@ -47,9 +48,8 @@ fn callback_gets_all_data_as_if_the_download_happened_all_at_once() {
         Some(&|msg| {
             match msg {
                 Event::ResumingPartialDownload => {
-                    let mut flag = callback_partial.lock().unwrap();
-                    assert!(!*flag);
-                    *flag = true;
+                    assert!(!callback_partial.load(Ordering::SeqCst));
+                    callback_partial.store(true, Ordering::SeqCst);
```

Comment:
Does this have the same side-effect of locking callback_partial for the duration of the read/assert/set sequence?

Reply:
[I think so](https://doc.rust-lang.org/nightly/std/sync/atomic/enum.Ordering.html#variant.SeqCst).